src/Core/Context.idr

```diff
@@ -1415,6 +1435,16 @@ hide fc n
               | res => ambiguousName fc n (map fst res)
          put Ctxt (record { gamma $= hideName nsn } defs)
 
+-- Set a name as Public that was previously hidden
+export
+unhide : {auto c : Ref Ctxt Defs} ->
```

Comment:
it would be nice, though I think not a blocker to merging this work, if there was a warning when unhiding a name that was not hidden to begin with.